manim/mobject/mobject.py

```diff
@@ -1232,6 +1232,9 @@ def scale(self, scale_factor: float, **kwargs) -> Self:
             The scaling factor :math:`\alpha`. If :math:`0 < |\alpha| < 1`, the mobject
             will shrink, and for :math:`|\alpha| > 1` it will grow. Furthermore,
             if :math:`\alpha < 0`, the mobject is also flipped.
+        scale_stroke
+            Boolean determining if the object's outline is scaled when the object is scaled.
+            If enabled, and object with 2px outline sclaed by a factor of .5 will have an outline of 1px.
```

Comment:
```suggestion
            If enabled, and object with 2px outline is scaled by a factor of .5 , it will have an outline of 1px.
```

Reply:
You are right, it makes way more sense to move it into `VMobject`. For some reason I couldn't get it to work last time, but now I have successfully overwritten `VMobject.scale()` to implement the stroke scaling functionality instead.

I implemented your changes to the documentation, and added a new example to demonstrate the parameter.

As for testing, I chose a unit test over graphic test. Testing the functionality of this chang is trivial with get_stroke_width().

Thank you for your feedback!

Reply:
Nice call on the unit test, you're right that's a better way to test it :)